test/OpenTelemetry.Instrumentation.AspNet.Tests/BasicTests.cs

```diff
@@ -14,4 +18,82 @@ public void AddAspNetInstrumentation_BadArgs()
         TracerProviderBuilder? builder = null;
         Assert.Throws<ArgumentNullException>(() => builder!.AddAspNetInstrumentation());
     }
+
+    [Theory]
+    [InlineData(true, false)]
+    [InlineData(false, false)]
+    [InlineData(true, true)]
+    [InlineData(false, true)]
+    public void SpansAndMetricsGeneratedOnlyWhenEnabled(bool tracesEnabled, bool metricsEnabled)
+    {
+        var activities = new List<Activity>();
+        var metrics = new List<Metric>();
+
+        var tracerProviderBuilder = Sdk.CreateTracerProviderBuilder();
+        if (tracesEnabled)
+        {
+            tracerProviderBuilder
+                .AddAspNetInstrumentation()
+                .AddInMemoryExporter(activities);
+        }
+
+        var meterProviderBuilder = Sdk.CreateMeterProviderBuilder();
+        if (metricsEnabled)
+        {
+            meterProviderBuilder = meterProviderBuilder
+                .AddAspNetInstrumentation()
+                .AddInMemoryExporter(metrics);
+        }
```

Comment:
I think that you need to always add InMemoryExporter otherwise collections are empty.
```suggestion
        var tracerProviderBuilder = Sdk.CreateTracerProviderBuilder()
            .AddInMemoryExporter(activities);

        if (tracesEnabled)
        {
            tracerProviderBuilder.AddAspNetInstrumentation();
        }

        var meterProviderBuilder = Sdk.CreateMeterProviderBuilder()
            .AddInMemoryExporter(metrics);

        if (metricsEnabled)
        {
            meterProviderBuilder.AddAspNetInstrumentation();
        }
```